src/component/strategies/hooks/useStrategyForm.ts

```diff
@@ -0,0 +1,84 @@
+import { ICustomStrategyParameter } from 'interfaces/strategy';
+import { useEffect, useState } from 'react';
+import useStrategies from '../../../hooks/api/getters/useStrategies/useStrategies';
+
+export const useStrategyForm = (
+    initialStrategyName: string = '',
+    initialStrategyDesc: string = '',
+    initialParams: ICustomStrategyParameter[] = []
+) => {
+    const [strategyName, setStrategyName] = useState(initialStrategyName);
+    const [strategyDesc, setStrategyDesc] = useState(initialStrategyDesc);
+    const [params, setParams] = useState(initialParams);
+    const [errors, setErrors] = useState({});
+    const { strategies } = useStrategies();
+
+    useEffect(() => {
+        setStrategyName(initialStrategyName);
+    }, [initialStrategyName]);
+
+    useEffect(() => {
+        setStrategyDesc(initialStrategyDesc);
+    }, [initialStrategyDesc]);
+
+    useEffect(() => {
+        setParams(initialParams);
+    }, [JSON.stringify(initialParams)]);
```

Comment:
Will the initial values change after the first call? If not, these effects cound be removed.

Reply:
These are used mainly for the Edit component, sometimes the response comes a bit late so we need a useEffect to update them.